lib/HLSL/DxilValidation.cpp

```diff
@@ -1658,8 +1658,12 @@ static void ValidateResourceDxilOp(CallInst *CI, DXIL::OpCode opcode,
   case DXIL::OpCode::CalculateLOD: {
     DxilInst_CalculateLOD lod(CI);
     Value *samplerHandle = lod.get_sampler();
-    if (GetSamplerKind(samplerHandle, ValCtx) != DXIL::SamplerKind::Default) {
-      ValCtx.EmitInstrError(CI, ValidationRule::InstrSamplerModeForLOD);
+    DXIL::SamplerKind samplerKind = GetSamplerKind(samplerHandle, ValCtx);
+    if (samplerKind != DXIL::SamplerKind::Default) {
+      // After SM68, Comparison is supported.
+      if (!ValCtx.DxilMod.GetShaderModel()->IsSM68Plus() ||
+          samplerKind != DXIL::SamplerKind::Comparison)
+        ValCtx.EmitInstrError(CI, ValidationRule::InstrSamplerModeForLOD);
```

Comment:
It looks like from the code path we could still get an error for SM6.8+ for a non-default sampler that is not a comparison sampler.

Is there some other sampler kind in hlsl that we could actually hit this error? If so it would be good to have test coverage for it.

Reply:
No real sampler kind left.
The rest sampler kinds are SamplerKind::Invalid and SamplerKind::Mono.
SamplerKind::Mono is only from dxbc convertor.